screens/map/MapScreen.js

```diff
@@ -35,198 +35,100 @@ import {
-      showDefaultStore: false,
-    };
-  }
+  const bottomSheetRef = useRef(null);
+  const mapRef = useRef(null);
+  const storeProducts = useStoreProducts(currentStore);
+  const { locationPermissions, initialLocation } = useCurrentLocation();
+  let stores = useStores();
 
-  async componentDidMount() {
-    // We get current location first, since we need to use the lat/lon found in _populateIntitialStoresProducts
-    await this._findCurrentLocation();
-    await this._populateInitialStoresProducts();
+  if (locationPermissions === 'granted') {
+    stores = orderStoresByDistance(initialLocation, stores);
   }
 
-  // TODO pretty high chance this should be either handled by navigation or `getDerivedStateFromProps`
-  // eslint-disable-next-line react/no-deprecated
-  componentWillReceiveProps(nextProps) {
-    const store = nextProps.route.params.currentStore;
-    const resetSheet = true;
-    store.distance = this.findStoreDistance(store);
-    this.changeCurrentStore(store, resetSheet);
-    const region = {
-      latitude: store.latitude,
-      longitude: store.longitude,
-      ...deltas,
-    };
-    this.setState({ region });
-  }
+  const showDefaultStore =
+    locationPermissions !== 'granted' ||
+    (stores.length > 0 && stores[0].distance > 100);
 
-  // Asks for permission if necessary, then gets current location
-  _findCurrentLocation = async () => {
-    const { status } = await Permissions.askAsync(Permissions.LOCATION);
-    // Error message not checked anywhere
-    if (status !== 'granted') {
-      this.setState({
-        locationErrorMsg: 'Permission to access location was denied',
-      });
-      Analytics.setUserProperty('location_permissions', 'denied');
-    } else {
-      try {
-        const location = await Location.getCurrentPositionAsync({
-          timeout: 3000,
-        });
-        const region = {
-          latitude: location.coords.latitude - deltas.latitudeDelta / 3.5,
-          longitude: location.coords.longitude,
-          ...deltas,
-        };
-        // Don't re-animate if we're using the default store
-        if (this._map && !this.state.showDefaultStore) {
-          this._map.animateToRegion(region, 1000);
-          this.setState({ locationErrorMsg: null });
-        } else {
-          this.setState({ locationErrorMsg: null, region });
-        }
-        Analytics.setUserProperty('location_permissions', 'granted');
-      } catch (err) {
-        console.log(err);
-        this.setState({
-          locationErrorMsg: 'Permission to access location was denied',
-        });
-        Analytics.setUserProperty('location_permissions', 'error');
-        logErrorToSentry({
-          screen: 'MapScreen',
-          function: '_findCurrentLocation',
-          error: err,
-        });
+  // This should only happen once on first load.
+  useEffect(() => {
+    if (
+      (!currentStore || !region) &&
+      locationPermissions &&
+      stores.length > 0
+    ) {
```

Comment:
This `useEffect` basically just sets the current store and region (map location) to defaults if the user has location disabled OR is over 100 miles away from the closest store.

Basically, it should only run:
- on the first load (`(!currentStore || !region)`)
- once locationPermissions has been set by `useCurrentLocation`
- once stores are loaded from `useStores`

Subsequent changing of current store/region is handled separately in `changeCurrentStore`.

The way I'm doing this feels really inelegant, but I'm struggling to figure out a better way to handle this. Thoughts?

Reply:
What if instead of using these conditions to set the `currentStore` and `region` with values in state, you allowed them to be empty/null and performed this handling as a "fallback" for those values? Would that be possible?